File: 1_sorting_algorithms/sorting_algorithms/py/selectionsort.py
import random
import time

from utils import swap


def selection_sort(arr):
    """O(n2) less swaps than bubble sort but still bad"""
    for i in range(len(arr)):
        minval = arr[i]
        minind = i
        for j in range(i + 1, len(arr)):
            if arr[j] < minval:
                minval = arr[j]
                minind = j
        swap(arr, minind, i)
    return arr


if __name__ == "__main__":
    n = 50000
    arr = [random.randint(0, n) for i in range(n)]

    start = time.time()
    selection_sort(arr)
    print(time.time() - start, "seconds")
    print(arr[:5])

File: 1_sorting_algorithms/sorting_algorithms/rs/src/algos/selectionsort.rs
pub fn selectionsort(arr: &mut Vec<i32>) {
    for i in 0..arr.len() {
        let mut minind = i;
        for j in i + 1..arr.len() {
            if arr[j] < arr[minind] {
                minind = j;
            }
        }
        arr.swap(i, minind);
    }
}
